geode-membership/src/integrationTest/java/org/apache/geode/distributed/internal/membership/gms/MembershipIntegrationTest.java

```diff
@@ -0,0 +1,206 @@
+    locator2.start();
+    final int locatorPort2 = locator2.getPort();
+
+    Membership<MemberIdentifier> membership2 =
+        createMembership(locator2, locatorPort1, locatorPort2);
+    start(membership2);
+
+    assertThat(membership1.getView().getMembers()).hasSize(2);
+    assertThat(membership2.getView().getMembers()).hasSize(2);
+  }
+
+  @Test
+  public void secondMembershipCanJoinUsingTheSecondLocatorToStart()
+      throws IOException, MemberStartupException {
+
+    final MembershipLocator<MemberIdentifier> locator1 = createLocator(0);
+    locator1.start();
+    final int locatorPort1 = locator1.getPort();
+
+    final Membership<MemberIdentifier> membership1 = createMembership(locator1, locatorPort1);
+    start(membership1);
+
+    final MembershipLocator<MemberIdentifier> locator2 = createLocator(0, locatorPort1);
+    locator2.start();
+    int locatorPort2 = locator2.getPort();
+
+    // Force the next membership to use locator2 by stopping locator1
+    locator1.stop();
+
+    Membership<MemberIdentifier> membership2 =
+        createMembership(locator2, locatorPort1, locatorPort2);
+    start(membership2);
+
+    assertThat(membership1.getView().getMembers()).hasSize(2);
+    assertThat(membership2.getView().getMembers()).hasSize(2);
+  }
+
+  private void start(final Membership<MemberIdentifier> membership)
+      throws MemberStartupException {
+    membership.start();
+    membership.startEventProcessing();
+  }
+
+  private Membership<MemberIdentifier> createMembership(
+      final MembershipLocator<MemberIdentifier> embeddedLocator,
+      final int... locatorPorts)
+      throws MembershipConfigurationException {
+    final boolean isALocator = embeddedLocator != null;
+    final MembershipConfig config = createMembershipConfig(isALocator, locatorPorts);
+
+    final MemberIdentifierFactoryImpl memberIdFactory = new MemberIdentifierFactoryImpl();
+
+    final TcpClient locatorClient =
+        new TcpClient(socketCreator, dsfidSerializer.getObjectSerializer(),
+            dsfidSerializer.getObjectDeserializer(), TcpSocketFactory.DEFAULT);
+
+    return MembershipBuilder.<MemberIdentifier>newMembershipBuilder(
+        socketCreator, locatorClient, dsfidSerializer, memberIdFactory)
+        .setMembershipLocator(embeddedLocator)
+        .setConfig(config)
+        .create();
+  }
+
+  private MembershipConfig createMembershipConfig(
+      final boolean isALocator,
+      final int[] locatorPorts) {
+    return new MembershipConfig() {
+      public String getLocators() {
+        return getLocatorString(locatorPorts);
+      }
+
+      // TODO - the Membership system starting in the locator *MUST* be told that is
+      // is a locator through this flag. Ideally it should be able to infer this from
+      // being associated with a locator
+      @Override
+      public int getVmKind() {
+        return isALocator ? MemberIdentifier.LOCATOR_DM_TYPE : MemberIdentifier.NORMAL_DM_TYPE;
```

Comment:
would it be better to send in the VM kind instead of the boolean `isALocator`? doing that would eliminate the conditional here.

Reply:
Seems like that would just shift the conditional? I think I'll leave it the way it is for now since we want to get rid of this flag entirely when we fix the membership code.